ext/date/php_date.c

```diff
@@ -3723,7 +3725,8 @@ PHP_METHOD(DateTimeZone, __wakeup)
 	myht = Z_OBJPROP_P(object);
 
 	if(php_date_timezone_initialize_from_hash(&return_value, &tzobj, myht) != SUCCESS) {
-		php_error_docref(NULL, E_ERROR, "Timezone initialization failed");
+		zend_throw_error(NULL, "Timezone initialization failed");
+		RETURN_FALSE;
```

Comment:
Technically, should this not be returning NULL instead? That's the standard way for at least constructors and other methods, when parameters can't be parsed. FALSE is a specific error's return.